### PR TITLE
Bump gl-native to v10.0.0, common to v20.0.0

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Bump gl-native to v10.0.0, common to v20.0.0.</changelog>`.

### Summary of changes

Bump gl-native to v10.0.0, common to v20.0.0.
Wrapping new native functions to switch projections will be handled in https://github.com/mapbox/mapbox-maps-android/pull/667.

### User impact (optional)

- some abstract classes like `Observer()`, `CustomLayerHost()`, `MapClient()`, `OfflineRegionObserver()` have become interfaces.

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/LegacyOfflineActivity.kt [hunk 1]
```diff
@@ -19,7 +19,7 @@ class LegacyOfflineActivity : AppCompatActivity() {
   private var mapView: MapView? = null
   private lateinit var binding: ActivityLegacyOfflineBinding
 
-  private val regionObserver: OfflineRegionObserver = object : OfflineRegionObserver() {
+  private val regionObserver: OfflineRegionObserver = object : OfflineRegionObserver {
```
Comment (alexander-kulikovskii) [inline app/src/main/java/com/mapbox/maps/testapp/examples/LegacyOfflineActivity.kt:22]: nit:
```suggestion
  private val offlineRegionObserver: OfflineRegionObserver = object : OfflineRegionObserver {
```

File: sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt [hunk 2]
```diff
@@ -332,19 +330,17 @@ class OfflineTest {
...
-        if (!data.cancelled && data.dataSource == "database" && data.request.kind == "tile") {
-          resourceRequests++
-        }
-        if (event.type == MapEvents.MAP_LOADING_ERROR) {
-          mapLoadingErrorCount++
-        }
-        if (event.type == MapEvents.MAP_LOADED || event.type == MapEvents.STYLE_LOADED) {
-          latch.countDown()
-        }
+    val observer = Observer { event ->
```
Comment (alexander-kulikovskii) [inline sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt:333]: duplicate [290-302]

File: sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt [hunk 3]
```diff
@@ -444,15 +438,13 @@ class OfflineTest {
...
-    val observer = object : Observer() {
-      override fun notify(event: Event) {
-        Logger.e(TAG, "type ${event.type}, data ${event.data.toJson()}")
-        if (event.type == MapEvents.MAP_IDLE) {
-          idleEventCount++
-        }
-        if (event.type == MapEvents.MAP_LOADING_ERROR) {
-          mapLoadingErrorCount++
-        }
+    val observer = Observer { event ->
```
Comment (alexander-kulikovskii) [inline sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt:441]: duplicate [408-416]

File: sdk/src/test/java/com/mapbox/maps/NativeMapTest.kt [hunk 4]
```diff
@@ -16,34 +16,22 @@ class NativeMapTest {
   fun subscribe() {
     val nativeMap = NativeMapImpl(map)
     val list = mutableListOf("foobar")
-    nativeMap.subscribe(
-      object : Observer() {
-        override fun notify(event: Event) {}
-      },
-      list
-    )
+    nativeMap.subscribe({ }, list)
```
Comment (alexander-kulikovskii) [inline sdk/src/test/java/com/mapbox/maps/NativeMapTest.kt:19]: nit:
```suggestion
    nativeMap.subscribe({}, list)
```

File: app/src/main/java/com/mapbox/maps/testapp/examples/LegacyOfflineActivity.kt (hunk 1 above)
Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/examples/LegacyOfflineActivity.kt:22]: I'll not be doing refactoring existent code as part of this PR, this PR is to adopt to latest core only.
We can revisit those nit's later as part of some other cleanup.

File: sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt (hunk 3 above)
Comment (kiryldz) [reply sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt:441]: I'll not be doing refactoring existent code as part of this PR, this PR is to adopt to latest core only.
We can revisit those nit's later as part of some other cleanup.

File: sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt (hunk 2 above)
Comment (kiryldz) [reply sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt:333]: I'll not be doing refactoring existent code as part of this PR, this PR is to adopt to latest core only.
We can revisit those nit's later as part of some other cleanup.

File: sdk/src/test/java/com/mapbox/maps/NativeMapTest.kt (hunk 4 above)
Comment (kiryldz) [reply sdk/src/test/java/com/mapbox/maps/NativeMapTest.kt:19]: `ktlint` is passing so let's leave as is, I see no difference tbh 😄

**Review by alexander-kulikovskii — APPROVED** [review]

**Review by Chaoba — APPROVED** [review]

File: sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt [hunk 5]
```diff
@@ -508,4 +508,8 @@ internal class NativeMapImpl(private val map: MapInterface) :
   override fun unsubscribe(observer: Observer) {
     map.unsubscribe(observer)
   }
+
+  override fun setMapProjection(value: Value) = map.setMapProjection(value)
+
+  override fun getMapProjection() = map.mapProjection
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt:514]: Don't we exposes these two API in this PR yet?

**Review by pengdev — APPROVED** [review]
LGTM

File: sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt [hunk 6]
```diff
@@ -287,19 +287,17 @@ class OfflineTest {
...
-        if (event.type == MapEvents.MAP_LOADED || event.type == MapEvents.STYLE_LOADED) {
-          latch.countDown()
-        }
+    val observer = Observer { event ->
+      Logger.e(TAG, "type ${event.type}, data ${event.data.toJson()}")
+      val data = event.getResourceEventData()
+      if (!data.cancelled && data.dataSource == "database" && data.request.kind == "tile") {
+        resourceRequests++
+      }
+      if (event.type == MapEvents.MAP_LOADING_ERROR) {
```
Comment (axti) [inline sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt:296]:
```suggestion
      when (event.type) {
         MapEvents.MAP_LOADING_ERROR -> mapLoadingErrorCount++
         MapEvents.MAP_LOADED, MapEvents.STYLE_LOADED -> latch.countDown()
       }
```

File: sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt [hunk 7]
```diff
@@ -508,4 +508,8 @@ internal class NativeMapImpl(private val map: MapInterface) :
   override fun unsubscribe(observer: Observer) {
     map.unsubscribe(observer)
   }
+
+  override fun setMapProjection(value: Value) = map.setMapProjection(value)
```
Comment (axti) [inline sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt:512]: Add type explicitly

File: sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt (hunk 5 above)
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt:514]: no, as mentioned in description, work will be done in https://github.com/mapbox/mapbox-maps-android/pull/667

File: sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt (hunk 7 above)
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt:512]: as mentioned in description, work will be done in #667

File: sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt (hunk 6 above)
Comment (kiryldz) [reply sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt:296]: I'll not be doing refactoring existent code as part of this PR, this PR is to adopt to latest core only.
We can revisit those nit's later as part of some other cleanup.

**Pull request merged**